### DISCUSSION TITLE
10 things that all WordPress plugin developers should avoid

### DISCUSSION THREAD

**Discussion by dartiss** [WordPress/developer-blog-content#136] [Topic Ideas]
This will be a post based on [the talk of the same name from WCEU 2023](https://europe.wordcamp.org/2023/session/10-things-that-all-wordpress-plugin-developers-should-avoid/). However, I will omit most screenshots where specific plugins are used (I don't want to finger point).

The 10 items will be...

1. Aggressive Promotion
2. Inconsistent and confusing menus
3. Don’t take over the user process
4. Avoid creating your own UI
5. Lack of data validation
6. Loading unnecessary content
7. Leaving content behind
8. Forgetting Accessibility Standards
9. Ignoring code quality checking
10. Getting lost in your translations

**Comment by bph** [comment]
Howdy, @dartiss 

[During the Editorial Group Meeting.](https://wordpress.slack.com/archives/C03RL47B3S8/p1688649990607389) we had quite a lively discussion about your proposal

Here are the bullet points: 

- Modify the title to something more positive (Best practices for Plugin developers) or  "Here are 10 things to do to make your plugin awesome!"
- It might help to connect with .org support forums moderators and Plugin review team to double-check on the list.
- Humor is fine, if self-owned,  ("six things I learned the hard way on the tennis hall of fame”)
- If possible, link each item back to a section in the plugin handbook.

How does this sound? What do you think? Doable? 
I'd be happy to make the connection to the support forum moderators or plugin review team.

**Comment by dartiss** [comment]
> Modify the title to something more positive

I always surprised my talk got through with this title, as I was always told to make titles positive. For some reason I forgot that rule this time, but it got through anyway. But, yes, I can switch that out.

> It might help to connect with .org support forums moderators and Plugin review team to double-check on the list

I did go to them before the original talk to ask for suggestions, but haven't yet put the results under their noses for review, so that would be a good idea. If I draft up my post in Google docs, I can let them take a look at it too, unless you think I should just keep it to the 10 items?

> Humor is fine, if self-owned, ("six things I learned the hard way on the tennis hall of fame”)

Yeah, for a talk humor is fine, but for this I'll keep it down. I want this to be more factual and precise.

> If possible, link each item back to a section in the plugin handbook.

Good idea. I'll do this.

So, all sounds good to me - let me know next steps.

**Comment by bph** [comment]
Awesome! @dartiss Thank you so much! For getting back so fast. 

>If I draft up my post in Google docs, I can let them take a look at it too, unless you think I should just keep it to the 10 items?
Oh, letting Plugin review team + support moderators take a look at a draft is a great idea and way to get more suggestions, :-) I don't think you need to keep it at 10 :-) 

Next Steps: 
We would need to get back to the Editorial Group and put it up again for approval. The next session is August 3th, 2023. I don't see a danger that it wouldn't get approved with the modifications you are prepared to make. 

For now, approval is 'pending'.
It's up to you, if you want to wait until you have certainty, or if you want to already start working on it, start collaborating with other teams etc. 

Either way, Google Doc is definitely the place to start out (we are all desperately waiting for the collaboration phase of the Gutenberg project)

**Comment by mburridge** [comment]
Hi @dartiss 

Thanks for proposing this topic and for being receptive to the suggestions offered. This proposal has been discussed again in [today's Editorial Board Meeting](https://wordpress.slack.com/archives/C03RL47B3S8/p1691067691366869), but we're still unable to reach a consensus.

We'd love it if you could re-work your proposal in the light of these suggestions. We really hope to be able to give you the go ahead for this post in the next meeting.

Please add your re-worded proposal as an additional comment to this thread. Thanks.

**Comment by dartiss** [comment]
> Please add your re-worded proposal as an additional comment to this thread.

The re-worded title? Is that what you mean? Or do you need a summary of it as well?

**Comment by mburridge** [comment]
The title is less important, that can be changed during the review process and often is. We'd like to see a new outline of the proposed post, especially showing the more positive slant that you intend to take.

**Comment by bph** [comment]
Hi  @dartiss, I looked back at how the meeting went, what Michael wrote to you and what's in the meeting notes. We were definitely sending entirely mixed messages. I apologize for that. That can't have been a pleasant experience for you. Sorry that I didn't stay on top of it better. 
<img width="678" alt="Screenshot 2023-09-05 at 12 35 07" src="https://github.com/WordPress/developer-blog-content/assets/39980/fb984d96-fd08-44ae-b382-a46b700afe99">

Please let me know if there is anything to assist you with this topic and ping me on Slack if you'd like to get on a call to discuss this in person.

**Comment by dartiss** [comment]
Thanks Birgit. I'm a little unsure as to what the current status of this is - your screenshot seems to show it's approved, but Michael's response made me think it wasn't yet.

**Reply by bph** [reply]
The meeting notes are the current status ✅

**Comment by bph** [comment]
Discussion closed. Continue conversation on issue 

- #163

**Comment by bph** [comment]
Published: [How WordPress developers can keep their users satisfied](https://developer.wordpress.org/news/2023/09/how-wordpress-developers-can-keep-their-users-satisfied/)
